app/code/Magento/InventoryCatalogAdminUi/Controller/Adminhtml/Inventory/BulkTransferPost.php

```diff
@@ -58,10 +141,15 @@ public function execute()
         $skus = $this->bulkSessionProductsStorage->getProductsSkus();
         $unassignSource = (bool) $this->getRequest()->getParam('unassign_origin_source', false);
 
+        $async = $this->bulkOperationsConfig->isAsyncEnabled();
+
         try {
-            $this->bulkInventoryTransfer->execute($skus, $originSource, $destinationSource, $unassignSource);
-            $this->messageManager->addSuccessMessage(__('Bulk inventory transfer was successful.'));
-        } catch (ValidationException $e) {
+            if ($async) {
+                $this->runAsynchronousOperation($skus, $originSource, $destinationSource, $unassignSource);
+            } else {
+                $this->runSynchronousOperation($skus, $originSource, $destinationSource, $unassignSource);
+            }
+        } catch (\Exception $e) {
```

Comment:
it's not secure to catch highest level exception `\Exception` and show its message to user
```
} catch (\Exception $e) {
            $this->messageManager->addErrorMessage($e->getMessage());
```

Just localized exception messages could be shown. So that you need to 
```
catch (\Exception $e) {
                 // Log exception here $e->getMessage()
                throw new LocalizedException(
                    __("Some user-friendly message to show")
                );
```